test/e2e/data/infrastructure-docker/v1beta1/main/clusterclass-quick-start.yaml

```diff
@@ -348,6 +348,7 @@ spec:
             criSocket: unix:///var/run/containerd/containerd.sock
             kubeletExtraArgs:
               eviction-hard: 'nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%'
+        files: []
```

Comment:
```suggestion
```

This change only works if the referenced templates contain a non-empty array. 
From Topology controllers view: an empty array here equals to not defining it, because of the `omitEmpty` [at the type definition](https://github.com/kubernetes-sigs/cluster-api/blob/main/bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go#L54).